src/System.IO/tests/TextReader/TextReaderTests.cs

```diff
@@ -0,0 +1,284 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class TextReaderTests
+    {
+        protected virtual char[] GetSmallData()
+        {
+            return "HELLO".ToCharArray();
+        }
+
+        protected virtual char[] GetLargeData()
+        {
+            var testData = "HELLO".ToCharArray();
+
+            var data = new List<char>();
+            for (int i = 0; i < 1000; i++)
+            {
+                data.AddRange(testData);
+            }
+
+            return data.ToArray();
+        }
+
+        protected (char[] chArr, CharArrayTextReader textReader) GetCharArray()
+        {
+            var chArr = new char[]
+            {
+                char.MinValue,
+                char.MaxValue,
+                '\t',
+                ' ',
+                '$',
+                '@',
+                '#',
+                '\0',
+                '\v',
+                '\'',
+                '\u3190',
+                '\uC3A0',
+                'A',
+                '5',
+                '\r',
+                '\uFE70',
+                '-',
+                ';',
+                '\r',
+                '\n',
+                'T',
+                '3',
+                '\n',
+                'K',
+                '\u00E6'
+            };
+
+            var tr = new CharArrayTextReader(chArr);
+
+            return (chArr, tr);
+        }
+
+        [Fact]
+        public void EndOfStream()
+        {
+            using (var tr = new CharArrayTextReader(GetSmallData()))
+            {
+                var result = tr.ReadToEnd();
+
+                Assert.Equal("HELLO", result);
+
+                Assert.True(tr.EndOfStream, "End of TextReader was not true after ReadToEnd");
+            }
+        }
+
+        [Fact]
+        public void NotEndOfStream()
+        {
+            using (var tr = new CharArrayTextReader(GetSmallData()))
+            {
+                var charBuff = new char[3];
+                var result = tr.Read(charBuff, 0, 3);
+
+                Assert.Equal(3, result);
+
+                Assert.Equal("HEL", new string(charBuff));
+
+                Assert.False(tr.EndOfStream, "End of TextReader was true after ReadToEnd");
+            }
+        }
+
+        [Fact]
+        public async Task ReadToEndAsync()
+        {
+            using (var tr = new CharArrayTextReader(GetLargeData()))
+            {
+                var result = await tr.ReadToEndAsync();
+
+                Assert.Equal(5000, result.Length);
+            }
+        }
+
+        [Fact]
+        public void TestRead()
+        {
+            var baseInfo = GetCharArray();
+            var tr = baseInfo.textReader;
```

Comment:
`using (var tr = baseInfo.textReader)` ?

Reply:
If `baseInfo` is still holding a reference to `textReader` will `using (var tr = baseInfo.textReader)` make a difference?

Scratched my head a lot on this when I looked at existing test cases for reference which were not doing the same. Please refer [here](https://github.com/dotnet/corefx/blob/3d0e156090e64093f4588e475b90679eabf6ed58/src/System.IO/tests/StreamReader/StreamReader.cs#L17)

Reply:
Thanks got it. I am not using the additional reference later on anyways. So doesn't make any difference even if it is invalidated. Between, something wrong with github. Didn't get any notification for your new comments on email and in between I just went ahead and made all the changes!!! So please re-review.

Thanks,
Mandar